### PR TITLE
migration to tauri beta

### PR DESCRIPTION
**Pull request by crapStone**
https://tauri.studio/en/docs/usage/guides/migration

### CODE REVIEW

**Comment by crapStone** [comment]
finished migration, but frontend build fails
@Epsilon02 fix it!

File: src/app/menubar/menubar.component.html [hunk 1]
```diff
@@ -6,8 +6,12 @@
     <mat-menu #menu="matMenu">
       <button mat-menu-item (click)="newDocument()">{{ strings.menu.newDocument }}</button>
       <button mat-menu-item (click)="loadFile()">{{ strings.menu.loadFile }}</button>
-      <button mat-menu-item (click)="exportFile('cb2')">{{ strings.menu.saveFile }}</button>
-      <button mat-menu-item (click)="exportFile('pdf')">{{ strings.menu.exportPDF }}</button>
+      <button mat-menu-item (click)="exportFile({ name: 'foo', extensions: ['cb2'] })">
+        {{ strings.menu.saveFile }}
+      </button>
+      <button mat-menu-item (click)="exportFile({ name: 'foo', extensions: ['cb2'] })">
```
Comment (6543) [inline src/app/menubar/menubar.component.html:12]:
```suggestion
      <button mat-menu-item (click)="exportFile({ name: 'foo', extensions: ['pdf'] })">
```

File: src/app/@core/services/tauri/tauri.service.ts [hunk 2]
```diff
@@ -1,15 +1,16 @@
 import { from, Observable } from 'rxjs';
 import { Injectable } from '@angular/core';
 
-import * as dialog from 'tauri/api/dialog';
-import * as tauri from 'tauri/api/tauri';
-import { open as tauriOpen } from 'tauri/api/window';
+// import * as dialog from 'tauri/api/dialog';
```
Comment (6543) [inline src/app/@core/services/tauri/tauri.service.ts:4]: rm comments?

File: package.json [hunk 3]
```diff
@@ -24,8 +24,9 @@
     "@angular/platform-browser": "~11.0.1",
     "@angular/platform-browser-dynamic": "~11.0.1",
     "@angular/router": "~11.0.1",
+    "@tauri-apps/api": "^1.0.0-beta-rc.3",
```
Comment (6543) [inline package.json:27]: rc.4 too?

Comment (crapStone) [reply package.json:27]: api was only rc.3

Comment (crapStone) [reply package.json:27]: https://www.npmjs.com/package/@tauri-apps/api

File: package.json [hunk 4]
```diff
@@ -42,6 +43,7 @@
     "@types/node": "^15.0.1",
     "@typescript-eslint/parser": "4.22.0",
     "@typescript-eslint/eslint-plugin": "4.22.0",
+    "@typescript-eslint/parser": "4.22.0",
```
Comment (6543) [inline package.json:46]: Dublcat

**Comment by 6543** [comment]
white window issue

**Comment by crapStone** [comment]
webpack issue

File: src/app/@core/services/tauri/tauri.service.ts [hunk 5]
```diff
@@ -1,7 +1,8 @@
 import { from, Observable } from 'rxjs';
 import { Injectable } from '@angular/core';
 
-import { dialog, shell, tauri } from '@tauri-apps/api';
+import { dialog, shell, tauri } from '@tauri-apps/api/index.js';
+import { InvokeArgs } from "@tauri-apps/api/tauri";
```
Comment (6543) [inline src/app/@core/services/tauri/tauri.service.ts:5]:
```suggestion
import { InvokeArgs } from ‘@tauri-apps/api/tauri’;
```

**Review by 6543 — APPROVED** [review]

**Review by kathi208 — APPROVED** [review]

**Comment by kathi208** [comment]
go for it

**Pull request merged**